wandb/sdk_py27/wandb_artifacts.py

```diff
@@ -202,6 +168,19 @@ def add_manifest_file(log_phy_path):
         termlog("Done. %.1fs" % (time.time() - start_time), prefix=False)
 
     def add_reference(self, uri, name=None, checksum=True, max_objects=None):
+        """adds `uri` to the artifact via a reference, located at `name`. 
+        You can use Artifact#get_path(`name`) after downloading
+        the artifact to retrieve this object.
+        
+        Arguments:
+        - `uri`:str - the URI path of the reference to add. Can be an object returned from
+            Artifact.get_path to store a reference to another artifact's entry.
+        - `name`:str - the path to save
+        """
+
+        # This is a bit of a hack, we want to check if the uri is a of the type
+        # ArtifactEntry which is a private class returned by Artifact.get_path in
+        # wandb/apis/public.py. If so, then recover the reference URL.
```

Comment:
It would be a programmer error if `ref_url_fn` were not `callable`, right? Maybe we should error in that case.

Reply:
Yes, that is true. Corrected in next commit